manifests/params.pp

```diff
@@ -41,6 +41,12 @@
     default => 'Package[fail2ban]',
   }
 
+  $before_file = $::osfamily ? {
```

Comment:
please use `$facts['os']['family']` instead. The new facts hash is preferred to access facts.